verilog/tools/preprocessor/verilog_preprocessor.cc

```diff
@@ -16,29 +16,44 @@
 #include <iostream>
 #include <string>
 
+#include "absl/flags/flag.h"
 #include "absl/flags/usage.h"
 #include "absl/status/status.h"
 #include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
+#include "common/lexer/token_stream_adapter.h"
 #include "common/util/file_util.h"
 #include "common/util/init_command_line.h"
 #include "common/util/subcommand.h"
+#include "verilog/analysis/flow_tree.h"
+#include "verilog/analysis/verilog_analyzer.h"
+#include "verilog/parser/verilog_lexer.h"
+#include "verilog/parser/verilog_token_enum.h"
+#include "verilog/preprocessor/verilog_preprocess.h"
 #include "verilog/transform/strip_comments.h"
 
 using verible::SubcommandArgsRange;
-using verible::SubcommandEntry;
 
-static absl::Status StripComments(const SubcommandArgsRange& args,
-                                  std::istream&, std::ostream& outs,
-                                  std::ostream&) {
-  if (args.empty()) {
-    return absl::InvalidArgumentError(
-        "Missing file argument.  Use '-' for stdin.");
+bool PrintGeneratedVariant(const verible::TokenSequence& recieved_variant,
+                           const int variants_counter,
+                           const bool is_variant_complete) {
+  if (!is_variant_complete) {
+    // Only accepts 2 variants at most. (to test the functionality).
+    // TODO(karimtera): How many variants are practical?
+    return variants_counter < 20;
   }
-  const char* source_file = args[0];
+  std::cout << "Variant number " << variants_counter + 1 << ":\n";
```

Comment:
Since we discussed above to probably not include the variant counter in the callback function, if you really want to have a counter, just count it separately, if you keep track of how often times you've been called

```c++
int counter = 0;
auto status = control_flow_tree.GenerateVariants([&counter](const verible::TokenSequence& variant) {
   ++counter;
   // Do whatever needs to be done to print
  return counter < kNumberOfVariantLimit;  // probably should come from a flag
});
```

Reply:
Sorry, I don't quite understand this syntax.
I'll just count variants inside `VariantReceiver` using a static local variable for now.

Reply:
Static counter sounds like a good first step, we can refine that later.

The syntax is a c++ closure, so the parameters and {} with return are the same as a function. As with closures in other language, it can also capture values from the surrounding scope; in this case, we capture a reference to the counter using the `[&counter]` part.